app/src/main/java/chat/rocket/android/activity/ServerConfigActivity.java

```diff
@@ -77,14 +83,27 @@ protected void onPause() {
   }
 
   private void onRenderServerConfigError(ServerConfig config) {
-    if (config != null) {
+    if (config.getState() == ServerConfig.STATE_CONNECTION_ERROR) {
       sessionObserver.unsub();
       showFragment(new RetryConnectFragment());
     } else {
       sessionObserver.sub();
     }
   }
 
+  private void continueWithServerConfigSession(final Session session) {
+    fetchPublicSettings()
+        .continueWith(task -> {
+          registerForPush();
+          return task;
+        })
+        .continueWith(task -> {
+          onRenderServerConfigSession(session);
+          return task;
+        })
+        .continueWith(new LogcatIfError());
+  }
```

Comment:
I see.

I think it is better to put `fetchPublicSettings` before registerListeners in RocketChatWebSocketThread, and create new observer (PushSettingsObserver) to observe whether push  is enabled in public settings and trigger `registerForPush` if the enabled setting is observed.